apps/react-world/src/components/login/LoginPageContainer.tsx

```diff
@@ -1,18 +1,30 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import LoginForm from './LoginForm';
 import LoginHeader from './LoginHeader';
+import useLogin from '@hooks/useLogin';
 
 const LoginPageContainer = () => {
-  const handleLogin = (data: { email: string; password: string }) => {
-    console.log('handleLogin: ' + JSON.stringify(data, null, 2));
-  };
+  const { loginError, loginStatus, handleLogin } = useLogin();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loginStatus === 'success') {
+      navigate('/'); // 홈페이지로 리다이렉트
+    }
+  }, [loginStatus, navigate]);
```

Comment:
로그인 여부에 따라 `리다이렉트`라면 `useLogin`에서 하나의 함수로 묶어줘도 좋을 것 같아요. 동일한 관심사지 않나 싶습니다! `handleValidAccess` 라는 네이밍은 마음에 안 들지만, 이런 형태로 useLogin이 핸들링하는 것은 어떻게 생각하시나요?

```suggestion
  const { loginError, loginStatus, handleValidAccess, handleLogin } = useLogin();

  handleValidAccess();
```

Reply:
음 이 부분은 코드 수정하면서 고민을 계속 해 보았는데, 제 선호이기는 하지만 저는 현재의 형태가 더 좋지 않을까 생각이 듭니다..! 아래 사항을 고려하였어요. 

### 1. 책임 분리 원칙 (Separation of Concerns):
useLogin 훅을 최초 구현했던 의도는 로그인 기능에 대한 책임이었습니다. (이름도 이걸 반영하기도 했구요)
리다이렉트는 로그인 후의 화면 플로우(=네비게이션)와 관련된 부분이므로, 컴포넌트나 다른 훅에서 처리하는 것이 더 좋지 않을까 싶었어요.
그리고 추후에 로그인이 필요한 다른 페이지나 모달에서 useLogin 훅을 사용할 경우 리다이렉트 '/' 가 아닌 곳으로 처리해야 할 수도 있을 것 같기도 합니다.

### 2. 재사용성과 확장성:
추후에 로그인 로직이 변경되어도 리다이렉트와 관련된 코드는 영향을 받지 않도록 하고 싶습니다.
또한, 다른 로그인 방식 (예: 소셜 로그인)이 추가되더라도 기존의 로그인 훅을 재사용하기 쉽도록 만드는 방향이 좋지 않을까 싶었습니다. (최소한의 로직만 담기는)
대신에 로그인 성공 여부나 상태를 반환해 이를 사용하는 컴포넌트에서 리다이렉트를 처리하는 것이 좋을 것이라 생각했습니다!